### PR TITLE
Adding ability to pass authenticator name and session data

### PR DESCRIPTION
**Pull request by kevintechie**
This change enables custom Torii authentication providers to use session data when the close() method is called.

### CODE REVIEW

File: addon/authenticators/torii.js [hunk 1]
```diff
@@ -103,12 +103,24 @@ export default BaseAuthenticator.extend({
...
+        }, () => {
+          delete this._provider;
+          reject();
+        });
+      } else {
         delete this._provider;
-        resolve();
-      }, reject);
+        reject();
+      }
```
Comment (marcoow) [inline addon/authenticators/torii.js]: Why this change?

File: addon/authenticators/torii.js [hunk 2]
```diff
@@ -103,12 +103,24 @@ export default BaseAuthenticator.extend({
     @return {Ember.RSVP.Promise} A promise that when it resolves results in the session being invalidated
     @public
   */
-  invalidate() {
+  invalidate(data) {
+    this._assertToriiIsPresent();
+
+    data = data || {};
     return new RSVP.Promise((resolve, reject) => {
-      this.get('torii').close(this._provider).then(() => {
```
Comment (marcoow) [inline addon/authenticators/torii.js]: I guess just adding `data` as a 2nd argument here would be sufficient?

Comment (kevintechie) [inline addon/authenticators/torii.js]: For my API, yes, it is sufficient to just add `data` as a parameter to the method and then pass it as a second argument in the `close()` method. However, it's possible that some APIs may return data in the response that may be needed by the custom Torii provider. I don't know if that's a realistic expectation.
